src/flows/pipes/RawFluxEditor/FunctionsList/DynamicFunctionsList.tsx

```diff
@@ -1,33 +1,68 @@
-import React, {FC} from 'react'
+import React, {FC, useCallback, useEffect} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
 
-import {FLUX_FUNCTIONS} from 'src/shared/constants/fluxFunctions'
-import {FluxToolbarFunction} from 'src/types/shared'
+// Components
+import {FluxFunction} from 'src/types/shared'
 import Fn from 'src/flows/shared/FilterList/InjectionOption'
 import FilterList from 'src/flows/shared/FilterList/FilterList'
-import FunctionTooltipContent from 'src/flows/pipes/RawFluxEditor/FunctionsList/perFunction/FunctionToolTipContent'
+import FluxDocsTooltipContent from 'src/flows/pipes/RawFluxEditor/FunctionsList/perFunction/FluxDocsTooltipContent'
 
+// Actions
+import {getFluxPackages} from 'src/shared/actions/fluxDocs'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+
+import {AppState} from 'src/types'
 interface Props {
-  onSelect: (fn: FluxToolbarFunction) => void
+  onSelect: (fn: FluxFunction) => void
+}
+
+const sortFuncs = (a, b) => {
+  if (a.package.toLowerCase() === b.package.toLowerCase()) {
+    return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
+  } else {
+    return a.package.toLowerCase() < b.package.toLowerCase() ? -1 : 1
+  }
 }
 
 const DynamicFunctionsList: FC<Props> = ({onSelect}) => {
+  const dispatch = useDispatch()
+  const fluxFunctions = useSelector(
```

Comment:
@TCL735 can i use useSelector like this? it feels odd but it works

Reply:
Yes, but turn that function inside the parentheses into a named function somewhere, perhaps in `src/shared/selectors/app.ts`